minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/C2NifiClientService.java

```diff
@@ -222,14 +231,18 @@ private C2OperationHandlerProvider c2OperationHandlerProvider(NiFiProperties niF
             new StandardFlowEnrichService(niFiProperties), flowPropertyEncryptor,
             StandardFlowSerDeService.defaultInstance(), niFiProperties.getProperty(FLOW_CONFIGURATION_FILE));
 
+        SyncResourceStrategy syncResourceStrategy =
+            new DefaultSyncResourceStrategy(resourceRepository, Path.of(c2AssetDirectory), niFiProperties.getNarAutoLoadDirectory().toPath());
```

Comment:
Is it a good idea to use the same asset dir for the asset sync and the asset push? Files could be overridden accidentally via the single asset push.

Reply:
Modified the logic to use 'repository' directory under the asset directory for the sync/resource asset artifacts